src/status_im/contexts/wallet/connected_dapps/view.cljs

```diff
@@ -15,33 +15,32 @@
     [utils.string]))
 
 (defn- on-disconnect
-  [wallet-account {:keys [name topic]}]
-  (rf/dispatch [:hide-bottom-sheet])
-  (rf/dispatch
-   [:wallet-connect/disconnect-dapp
-    {:topic      topic
-     :on-success (fn []
-                   (rf/dispatch [:toasts/upsert
-                                 {:id   :dapp-disconnect-success
-                                  :type :positive
-                                  :text (i18n/label :t/disconnect-dapp-success
-                                                    {:dapp    name
-                                                     :account (:name wallet-account)})}]))
-     :on-fail    (fn []
-                   (rf/dispatch [:toasts/upsert
-                                 {:id   :dapp-disconnect-failure
-                                  :type :negative
-                                  :text (i18n/label :t/disconnect-dapp-fail
-                                                    {:dapp    name
-                                                     :account (:name wallet-account)})}]))}]))
+  [{:keys [name url topic]}]
+  (let [dapp-name (core/compute-dapp-name name url)]
+    (rf/dispatch [:hide-bottom-sheet])
+    (rf/dispatch
+     [:wallet-connect/disconnect-dapp
+      {:topic      topic
+       :on-success (fn []
+                     (rf/dispatch [:toasts/upsert
+                                   {:id   :dapp-disconnect-success
+                                    :type :positive
+                                    :text (i18n/label :t/disconnect-dapp-success
+                                                      {:dapp dapp-name})}]))
+       :on-fail    (fn []
+                     (rf/dispatch [:toasts/upsert
+                                   {:id   :dapp-disconnect-failure
+                                    :type :negative
+                                    :text (i18n/label :t/disconnect-dapp-fail
```

Comment:
Out of curiosity @clauxx, I couldn't find in Figma the toast with the message `Failed to disconnect ...` I'm looking around here https://www.figma.com/design/HOJ4lV2ko3jzMjorsDzaoP/Wallet-%2F-dApp-notifications?node-id=23-40254&t=lev3hBIhvC7I0N1d-0

Is this a fallback toast, but which is not covered in Figma?

Reply:
It is a fallback in case disconnection fails (so not on figma), which ideally should be very rare, but still nice to give the user some feedback. Do you think it should be added there?

Reply:
It's really nice that you cared about this 💯 

In general I think it's good @clauxx, because failure conditions should be part of the UX, not an after thought that devs implement, even if they are rare. Being rare can still be a cause of failure for many/important users (e.g. 0.01% daily rate but with 100k users could be 10 pissed off users every day if the error is more critical).

Sometimes it can be difficult for a designer to anticipate failure conditions, but that's where we can work with them more closely or involve them in the loop during development.

Reply:
Yeah, I think after the release will make sense to fully align the user flows with the implementation, so all these error states are considered.